packages/wdio-allure-reporter/src/index.ts

```diff
@@ -1,605 +1,686 @@
-    }
-
-    /**
-     * Add attachment
-     * @name addAttachment
-     * @param {string} name         - attachment file name
-     * @param {*} content           - attachment content
-     * @param {string=} mimeType    - attachment mime type
-     */
-    static addAttachment = (name, content, type) => {
-        if (!type) {
-            type = content instanceof Buffer ? 'image/png' : typeof content === 'string' ? 'text/plain' : 'application/json'
-        }
-        tellReporter(events.addAttachment, { name, content, type })
-    }
-
-    /**
-     * Start allure step
-     * @name startStep
-     * @param {string} title - step name in report
-     */
-    static startStep = (title) => {
-        tellReporter(events.startStep, title)
-    }
-
-    /**
-     * End current allure step
-     * @name endStep
-     * @param {StepStatus} [status='passed'] - step status
-     */
-    static endStep = (status = stepStatuses.PASSED) => {
-        if (!Object.values(stepStatuses).includes(status)) {
-            throw new Error(`Step status must be ${Object.values(stepStatuses).join(' or ')}. You tried to set "${status}"`)
-        }
-        tellReporter(events.endStep, status)
-    }
-
-    /**
-     * Create allure step
-     * @name addStep
-     * @param {string} title - step name in report
-     * @param {Object} [attachmentObject={}] - attachment for step
-     * @param {string} attachmentObject.content - attachment content
-     * @param {string} [attachmentObject.name='attachment'] - attachment name
-     * @param {string} [attachmentObject.type='text/plain'] - attachment type
-     * @param {string} [status='passed'] - step status
-     */
-    static addStep = (title, { content, name = 'attachment', type = 'text/plain' } = {}, status = stepStatuses.PASSED) => {
-        if (!Object.values(stepStatuses).includes(status)) {
-            throw new Error(`Step status must be ${Object.values(stepStatuses).join(' or ')}. You tried to set "${status}"`)
-        }
-
-        const step = content ? { title, attachment: { content, name, type }, status } : { title, status }
-        tellReporter(events.addStep, { step })
-    }
-
-    /**
-     * Add additional argument to test
-     * @name addArgument
-     * @param {string} name - argument name
-     * @param {string} value - argument value
-     */
-    static addArgument = (name, value) => {
-        tellReporter(events.addArgument, { name, value })
-    }
-}
-
-export default AllureReporter
+import WDIOReporter from '@wdio/reporter'
+import Allure from 'allure-js-commons'
+import Step from 'allure-js-commons/beans/step'
+import { getTestStatus, isEmpty, tellReporter, isMochaEachHooks, getErrorFromFailedTest, isMochaAllHooks, getLinkByTemplate } from './utils'
+import { events, PASSED, PENDING, SKIPPED, stepStatuses, testStatuses } from './constants'
+
+/**
+ * When you add a new option, please also update the docs at ./packages/wdio-allure-reporter/README.md

```

Comment:
@christian-bromann Added this. The previous generated typings does not automatically update the docs in README.md, so this is not any worse